### ISSUE TITLE
Android: fs.writeFile fails to set file mode

### ISSUE THREAD

**Issue by obastemur** [jxcore/jxcore#318]
`simple/test-fs-write-file.js` fails on Android with the output below;

```
writing to /mnt/sdcard/jxcore/tmp/test.txt
writing to /mnt/sdcard/jxcore/tmp/test2.txt
writing to /mnt/sdcard/jxcore/tmp/test3.txt
file written
file2 written
done

assert.js:100
  throw new assert.AssertionError({
AssertionError: 448 == 384
    at /mnt/sdcard/jxcore/simple/test-fs-write-file.js:68:12
    at Object.oncomplete (fs.js:83:15)
```

Possible this is something not directly related to fs.writeFile. `test-fs-chmod.js` also fails.

**Comment by ktrzeciaknubisa** [comment]
I've found some possible explanations:

1. sdcard is FAT formatted, which doesn't really support permisions. Running from shell `chmod XXX file` runs without error, but does not really change the mode as I saw it. [link](http://electron.mit.edu/~gsteele/linuxfaq/index.html#2)
2. sdcard is NTFS formatted, but is mounted with "-noexec" flag by default. [link](http://stackoverflow.com/questions/26050673/how-to-adb-chmod-750-on-sdcard-using-batch)

I had similar case and I couldn't even set 777 mode for jx while it was located under sdcard.

So it would be enough to move the test files out of sdcard. I've tested it under `/data/local/tmp/` and they both run OK.
